### PR TITLE
Update taxonomy sidebar design

### PR DESCRIPTION
**Pull request by steventux**
https://trello.com/c/zzot5dZq/157-taxonomy-sidebar-add-link-to-collections
https://trello.com/c/NtD5rC4S/121-taxonomy-sidebar

screenshot from 2017-12-12 16-36-52

This combines the changes from https://github.com/alphagov/static/pull/1200 to make the taxonomy sidebar styles similar to the related-navigation component and also the feature of rendering collections links below the taxon links.

### CODE REVIEW

File: app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss [hunk 1]
```diff
@@ -1,29 +1,48 @@
...
+    @include bold-19;
   }
 
   .taxon-description {
-    margin-bottom: 0.75em;
+    margin-top: 0;
+    margin-bottom: $gutter-one-third;
+  }
+
+  a {
```
Comment (maxgds) [inline app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss]: We usually wouldn't style an "a" directly. Sometimes we are forced to, but we absolutely need to ensure we're not styling all a tags on the page, so 
.taxon-description a { blah } 
or something may be necessary - ideally though the anchor would have a styling class such as .govuk-taxonomy-sidebar__link

File: app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss [hunk 2]
```diff
@@ -1,29 +1,48 @@
...
+    text-decoration: none;
+  }
+
+  nav {
```
Comment (maxgds) [inline app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss]: again, this needs to be scoped

File: app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss [hunk 3]
```diff
@@ -1,29 +1,48 @@
...
+    margin-bottom: $gutter;
+  }
+
+  li {
```
Comment (maxgds) [inline app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss]: scoping needed

File: app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss [hunk 4]
```diff
@@ -1,29 +1,48 @@
...
+    padding: 0;
+    list-style-type: none;
+    margin-bottom: $gutter-one-third;
   }
 
   ul {
```
Comment (maxgds) [inline app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss]: scope

File: app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss (hunk 1 above)
Comment (nickcolley) [reply app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss]: These seem to be nested within the selector at the top, but massive 👍 to always avoiding styling base elements if possible.

We should only be doing this if there's no other way as it directly conflicts with our naming conventions that allow us to have confidence with components across so many applications.

https://github.com/alphagov/govuk_publishing_components/blob/master/docs/component_conventions.md

Comment (nickcolley) [reply app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss]: Specifically we're using [BEM](https://github.com/alphagov/govuk_publishing_components/blob/master/docs/component_conventions.md#bem) and [namespacing](https://github.com/alphagov/govuk_publishing_components/blob/master/docs/component_conventions.md#namespacing)

**Comment by steventux** [comment]
@maxgds @nickcolley I'm confused about requested changes now, could you clarify what, if anything, needs changing on this PR please?

**Comment by maxgds** [comment]
I hadn't noticed the things I commented on were nested, as such they are they are safe to go out.

However, we shouldn't really be applying the css directly to elements even when embedded. I'd say mark it as tech debt, put it on the board in the tech debt column Vanita created and let this go out as-is for now.

**Comment by maxgds** [comment]
The diff is confusing - Steve hasn't actually introduced this, hence my comment. And it's relatively safe as it's nested in the scss, not completely bare.

**Comment by steventux** [comment]
@nickcolley forgive my ignorance, but it's not clear which changes are necessary to make this PR acceptable production-ready component code.
It seems like some unconventional CSS scoping from previous work is being conflated with the intended scope of this PR (some cursory design changes and adding the collections links, none of which add unscoped styles).
Is the fixing of the unconventional CSS a blocker to any other work on this component? If so can you detail what remedial work needs doing please? I'm not certain what _we should just make the updates in this review_ means in terms of fixing scope.

**Comment by nickcolley** [comment]
@steventux if we're writing new CSS we should be using the correct naming conventions (linked above). Old code is fine, and is covered by https://trello.com/c/SCio6yb4/45-static-components-need-updating-to-new-component-principles

**Comment by steventux** [comment]
My bad - had cherry-picked a commit from https://github.com/alphagov/static/pull/1200, was unaware this contained the offending CSS

**Comment by steventux** [comment]
@nickcolley @maxgds hopefully this https://github.com/alphagov/static/pull/1222/commits/fa16a2138443603c9027512fabb8b7b3345fc766 sorts out the element selectors. Let me know if I'm off with the BEM naming.

**Comment by steventux** [comment]
@nickcolley I've amended the component styles so they follow the BEM naming convention, apologies @maxgds for not understanding the original review comments.
Anything else to do on this PR?

File: app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss [hunk 5]
```diff
@@ -1,29 +1,52 @@
 .govuk-taxonomy-sidebar {
-  border-top: 10px solid $mainstream-brand;
-  padding-top: 5px;
   @include core-16;
+  border-top: 2px solid $govuk-blue;
+  padding-top: $gutter-half;
+}
+
+.govuk-taxonomy-sidebar__related-heading,
```
Comment (nickcolley) [inline app/assets/stylesheets/govuk-component/_taxonomy-sidebar.scss]: Non blocking:

These classes look to be the same, if we decouple how we present parts of this component from the semantics of this component, could these be change to a single '.govuk-taxonomy-sidebar__heading`?

**Comment by steventux** [comment]
Thanks @nickcolley your comment makes sense to me, I've renamed the heading and link classes to simplify: https://github.com/alphagov/static/pull/1222/commits/dd4211811e9342ba8d5952d256b8655ef2ff1303#diff-b2af3eb92cd75ba5d8c0d1ab14ce7424R7
I will think of a better name for the collections classes as it suffers from the same naming problem.

**Pull request merged**